### DISCUSSION TITLE
Relational placeholders

### DISCUSSION THREAD

**Discussion by florensm** [BG-Software-LLC/SuperiorSkyblock2#1827] [Pending Feature Requests]
It would be nice to have a placeholder which returns yes or no depending if someone is on your team or not. (Or green color for yes and red for no)

This would be helpful in pvp and other chaotic events.

I had this code before which worked fine for a while but i then updated my server from 1.8 go 1.20 and it stopped working... The owner of the code said i could give it to the developers of this plugin.

Java
package me.MrGraycat.IslandPlaceholder;

import org.bukkit.entity.Player;

import com.bgsoftware.superiorskyblock.api.SuperiorSkyblockAPI;
import com.bgsoftware.superiorskyblock.api.island.Island;
import com.bgsoftware.superiorskyblock.api.wrappers.SuperiorPlayer;

import me.clip.placeholderapi.expansion.PlaceholderExpansion;
import me.clip.placeholderapi.expansion.Relational;

public class RelIslandPlaceholder extends PlaceholderExpansion implements Relational {
    @Override
    public String getAuthor() {
        return "Mr_Graycat";
    }

    @Override
    public String getIdentifier() {
        return "ssisland";
    }

    @Override
    public String getVersion() {
        return "1.0";
    }
    
    @Override
    public String onPlaceholderRequest(Player player1, Player player2, String identifier) {
        if (player1 == null || player2 == null)
            return "";
        
        SuperiorPlayer SPlayer1 = SuperiorSkyblockAPI.getPlayer(player1);
        SuperiorPlayer SPlayer2 = SuperiorSkyblockAPI.getPlayer(player2);
        Island island = SPlayer1.getIsland();
        
        if (island == null)
            return "";
        
        if (island.getIslandMembers(true).contains(SPlayer2))
            return "§a";
        return "";
    }
}

This way you could just put a specific placeholder in front of a nametag or tablist name and it would color the name red or green.

Code above was a placeholderapi addon for the plugin TAB: https://github.com/NEZNAMY/TAB

**Comment by Tuogptuan** [comment]
https://bg-software.com/superiorskyblock/api/com/bgsoftware/superiorskyblock/api/island/Island.html#isMember-com.bgsoftware.superiorskyblock.api.wrappers.SuperiorPlayer-
